src/entities/functionality-config.ts

```diff
@@ -0,0 +1,15 @@
+/**
+ * Flags used to enable or disable certain features in the sdk.
+ * @public
+ */
+export interface FlagsConfig {
```

Comment:
Hmm I'm not sure if I like the name... I guess it's too generic 😅... If we want to follow more the mobile SDKs pattern, I would say we should change the `configure` method to accept an object (something like `PurchasesConfiguration`), which can have all the required and optional parameters, then make `collectAutomaticallyUTMParamsAsMetadata` one of those top-level parameters with a default value. I know this would increase the number of parameters in that `PurchasesConfiguration` object, but it would avoid this very generic level of indirection. Wdyt?

Note: If we end up with that route, we should keep the existing configure function but deprecate it, so we don't make a breaking change.

Reply:
I don't have a strong opinion.
I'd also like to change towards a `configure` that has direct named parameters, however I think is too big for the context of this PR.

Considering that in any case we would need to deprecate this method I think it's fine to keep it as I proposed and make that change as a follow-up PR.
I think having the possibility to opt out is important for this feature.

Reply:
PS: I just renamed the file to keep it consistent with the class name

Reply:
Hmm still feels like a weird API to me TBH... but as long as we are planning to move to a configuration with an object in the future, and not expand on this, I wouldn't be too opposed 👍